src/mgba_printf/mini_printf.c

```diff
@@ -79,13 +156,43 @@ static s32 _puts(char *s, s32 len, void *buf)
     return b->pbuffer - p0;
 }
 
+static s32 _puts2(char *s, s32 len, void *buf)
+{
+    char *p0;
+    s32 i;
+    struct mini_buff *b;
+
+    if (!buf) 
+        return len;
+    
+    b = buf;
+    p0 = b->buffer;
+
+    /* Copy to buffer */
+    for (i = 0; i < len; i++) {
+        if(b->pbuffer == b->buffer + b->buffer_len - 1) {
+            break;
+        }
+        *(b->pbuffer ++) = decodeChar(s[i]);
+    }
+    *(b->pbuffer) = 0;
+    return b->pbuffer - p0;
+}
+
 static s32 mini_strlen(const char *s)
 {
 	s32 len = 0;
 	while (s[len] != '\0') len++;
 	return len;
 }
 
+static s32 mini_strlen2(const char *s)
```

Comment:
cryptic name, also essentially a copy of the library function `StringLength` that already exists in the game. It is debatable wether this should work as a standalone, but if not this should just use the string library function of the game.